### ISSUE TITLE
Requests in queue

### ISSUE THREAD

**Issue by piernik** [mgonto/restangular#1032]
In my app I'm using resolve func that sometimes has to make couple server requests (4-6) for different data. That requests are with restangular.
Problem is that my server die with 4 or more requests in the same time.
What is solution? Is there in restangular option that it queues requests or delays them a bit (ex. 50 ms)?

**Comment by TimDaub** [comment]
As Restangular uses promises as a callback mechanism, you could for example use angular's build-in `$timeout` to delay execution...

I found something for you here: http://bahmutov.calepin.co/promises-with-timeouts.html

**Comment by grabbou** [comment]
Yeah, @TimDaub is correct. Generally any kind of library for throttling would do the trick -> first example one from Google -> https://www.npmjs.com/package/promise-throttle. Hope that's going to work for you :)

**Comment by piernik** [comment]
I found perfect solution - it's app config
```
$httpProvider.interceptors.push(['$q', function ($q) {
                var _queue = [];

                /**
                 * Executes the top function on the queue (if any).
                 */
                function _executeTop() {
                    if (_queue.length === 0) {
                        return;
                    }
                    _queue[0]();
                }

                return {
                    /**
                     * Blocks each request on the queue. If the first request, processes immediately.
                     */
                    request: function (config) {
                        if (config.url.substring(0, 4) == 'http') {
                            var deferred = $q.defer();
                            _queue.push(function () {
                                deferred.resolve(config);
                            });
                            if (_queue.length === 1) {
                                _executeTop();
                            }
                            return deferred.promise;
                        } else {
                            return config;
                        }
                    },
                    /**
                     * After each response completes, unblocks the next request.
                     */
                    response: function (response) {
                        if (response.config.url.substring(0, 4) == 'http') {
                            _queue.shift();
                            _executeTop();
                        }
                        return response;
                    },
                    /**
                     * After each response errors, unblocks the next request.
                     */
                    responseError: function (responseError) {
                        if (responseError.config.url.substring(0, 4) == 'http') {
                            _queue.shift();
                            _executeTop();
                        }
                        return $q.reject(responseError);
                    },
                };
            }]);
```
